Doc/reference/expressions.rst

```diff
@@ -1054,10 +1054,20 @@ used in the same call, so in practice this confusion does not often arise.
 
 If the syntax ``**expression`` appears in the function call, ``expression`` must
 evaluate to a :term:`mapping`, the contents of which are treated as
-additional keyword arguments.  If a keyword is already present
-(as an explicit keyword argument, or from another unpacking),
+additional keyword arguments. If a parameter matching a key has already been
+given a value (by an explicit keyword argument, or from another unpacking),
 a :exc:`TypeError` exception is raised.
 
+When ``**expression`` is used, each key in this mapping must be
+an instance of :class:`str`.
```

Comment:
I wonder if we should be vague here and write "a string" ? The SC said that the issue of subclasses of strings was still undefined.

Reply:
"As instance of :class:`str`" could just as easily be interpreted as `isinstance(key, str)` so I'd say it is already underspecified enough. 🤷🏾

Reply:
I'm happy with either if core devs can agree.

The more vague the language here, the greater the need for precision in a subsequent sentence, the glossary or elsewhere. Petr's statement about sub-classes on Discourse was good, but a bit long to do it justice here. I have an idea, but it may be another issue/PR.

Reply:
My point was that Petr's post indicated the following:

- If its type is exactly `str` this is allowed by the language spec
- If its type is a subclass of `str` then it is up to the implementation
- If its type is anything else then it is disallowed

Writing "instance of `str`" that makes the second bullet seem like the spec allows it so a conforming implementation must allow it. So, short of explicit indicating undefined behavior, I figured we'd write "must be a string" which clearly allows the first bullet, clearly disallows the third, and is vague enough about the second.